packages/jellyfish-api-core/__tests__/category/loan/updateLoanToken.test.ts

```diff
@@ -0,0 +1,289 @@
+    await testing.rpc.loan.updateLoanToken('Token1', { name: 'x'.repeat(129) }) // 129 letters
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].name).toStrictEqual('x'.repeat(128)) // Only remain the first 128 letters.
+  })
+
+  it('should updateLoanToken if two loan tokens have the same name', async () => {
+    const priceFeedId2 = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token5',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'Token5',
+      priceFeedId: priceFeedId2
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.updateLoanToken('Token5', { name: 'Token1' }) // Same name as Token1's name
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].name).toStrictEqual('Token1')
+  })
+
+  it('should updateLoanToken with given priceFeedId', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token2',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    await testing.rpc.loan.updateLoanToken('Token1', { symbol: 'Token2', priceFeedId })
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    expect(data[loanTokenId].priceFeedId).toStrictEqual(priceFeedId)
+  })
+
+  it('should not updateLoanToken if priceFeedId is invalid', async () => {
+    const promise = testing.rpc.loan.updateLoanToken('Token1', { priceFeedId: 'e40775f8bb396cd3d94429843453e66e68b1c7625d99b0b4c505ab004506697b' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanUpdateLoanTokenTx execution failed:\noracle (e40775f8bb396cd3d94429843453e66e68b1c7625d99b0b4c505ab004506697b) does not exist!\', code: -32600, method: updateloantoken')
+  })
+
+  it('should updateLoanToken if mintable is true', async () => {
+    await testing.rpc.loan.updateLoanToken('Token1', { mintable: true })
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].mintable).toStrictEqual(true)
+  })
+
+  it('should updateLoanToken if mintable is false', async () => {
+    await testing.rpc.loan.updateLoanToken('Token1', { mintable: false })
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].mintable).toStrictEqual(false)
+  })
```

Comment:
probably can make a single test with name `should updateLoanToken mintable`
then first you can set it to `false` since it is already in `true`. then make it to `true` again.